### PR TITLE
Fix for #4633 - use PSS padding for SHR POP memory crypto provider

### LINKED ISSUE #4633: [Bug] Stop using PKCS1 padding in POP+SHR in-memory crypto provider
### Library version used

4.59

### .NET version

All

### Scenario

ConfidentialClient - web site (AcquireTokenByAuthCode), ConfidentialClient - web api (AcquireTokenOnBehalfOf), ConfidentialClient - service to service (AcquireTokenForClient)

### Is this a new or an existing app?

None

### Issue description and reproduction steps

Compliance bug - CodeQL has flagged the usage of a deprecated padding scheme. See https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2827295?src=WorkItemMention&src-action=artifact_link for details

### Relevant code snippets

_No response_

### Expected behavior

_No response_

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

_No response_

### Solution and workarounds

_No response_

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #4633

Existing integration tests cover this well. Took me quite some time to fix them :)

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs [hunk 1]
```diff
@@ -27,7 +28,14 @@ public InMemoryCryptoProvider()
 
         private void InitializeSigningKey()
         {
+#if NETFRAMEWORK
+            // This method was obsolete in .NET,
+            // but Create() on .NET FWK defaults to PKCS1 padding.
+            _signingKey = RSA.Create("RSAPSS");
+#else
             _signingKey = RSA.Create();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs:36]: What's the default for .NET Standard? Should we explicitly specify RSAPSS for all targets, not rely one default?

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs:36]: The method `RSA.Create("RSAPSS")` is obsolete (returns NULL) on .NET and .NETSTANDARD. On these tfms, `RSA.Create()` factory returns a modern implementation of RSA, which uses PSS padding by default. 

On .NET FWK however, `RSA.Create()` returns a old implementation of RSA (RsaCryptoProvider) which uses PKCS1 padding by default. `RSA.Create("RSAPSS")` is NOT deprecated on .NET FWK.

I didn't find a way to change padding after the RSA object is created. The RSA object is read-only. So it has to be done via the ctor.

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs [hunk 2]
```diff
@@ -36,7 +44,10 @@ private void InitializeSigningKey()
 
         public byte[] Sign(byte[] payload)
         {
-            return _signingKey.SignData(payload, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+            return _signingKey.SignData(
+                payload,
+                HashAlgorithmName.SHA256,
+                RSASignaturePadding.Pss);
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs:50]: will this affect generic authority? i.e. if the authority does not support PSS padding

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs:50]: POP support for non-AAD authorities is not guarateed. POP SHR API is anyway experimental.

**Comment by bgavrilMS** [comment]
@gladjohn - thanks for your review. Please formally "Accept" or "Reject" the PR to speed things up. If you have minor comments, consider "Accept" which implies that you don't have big concerns and it's ok for the PR to be merged once comments are resolved (similar to ADO's "accept with comments").

**Pull request merged**